nixos/modules/services/databases/postgresql.nix

```diff
@@ -361,12 +361,25 @@ in
       };
 
       enableTCPIP = mkOption {
-        type = types.bool;
-        default = false;
+        type = types.nullOr types.bool;
+        default = null;
         description = ''
-          Whether PostgreSQL should listen on all network interfaces.
-          If disabled, the database can only be accessed via its Unix
-          domain socket or via TCP connections to localhost.
+          Whether PostgreSQL should listen on network interfaces.
+          If 'false', the database can only be accessed via its Unix
+          domain socket.
+          A value of 'null' defaults to 'true'.
+        '';
+      };
+
+      listenAddresses = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          The TCP/IP address(es) on which the server is to listen for connections from client applications.
+          The value takes the form of a comma-separated list of host names and/or numeric IP addresses.
+          The special entry * corresponds to all available IP interfaces.
+
+          See the PostgreSQL documentation on [listen_address](https://www.postgresql.org/docs/16/runtime-config-connection.html#GUC-LISTEN-ADDRESSES).
         '';
```

Comment:
It would be clearer if it was a list of strings later joined with comma in implementation

Reply:
I agree with this, however I am not sure whether the Nixpkgs policy is to map the settings of the services as close as possible ("The value takes the form of a comma-separated list of host names and/or numeric IP addresses")  or to provide a more abstract interface.

On a similar topic, other packages use alternative names for this such as `hosts`, `bind`, `listen`, `interface`, `ports`, ... `listen_addresses` is the name of the setting in PostgreSQL (and a few other services).

Reply:
I would prefer update this branch with a list of strings (or IP addresses if there is a type for that)  if that is compatible with the Nixpkgs policy.